__tests__/api-test.test.js

```diff
@@ -160,10 +164,77 @@ describe("/api/articles/:article_id/comments", () => {
       .get('/api/articles/999/comments')
       .expect(404)
       .then(({ body }) => {
-        const {msg} = body
+        const { msg } = body
         expect(msg).toBe('article does not exist');
       });
 
   });
   
-});
+
+  describe('/api/articles/:article_id/comments', () => {
+    test("POST:201 inserts a new comment to the database and sends the new comment back to the user", () => {
+      const newComment = {
+        username: "rogersop",
+        body: "test body comment",
+      };
+      return request(app)
+        .post('/api/articles/1/comments').send(newComment).expect(201)
+        .then(({ body }) => {
+          const { comment } = body;
+          expect(comment).toHaveProperty("comment_id", 19);
+          expect(comment).toHaveProperty("votes", 0);
+          expect(comment).toHaveProperty("created_at", (expect.any(String)));
+          expect(comment).toHaveProperty("author", "rogersop");
+          expect(comment).toHaveProperty("body", "test body comment");
+          expect(comment).toHaveProperty("article_id", 1);
+        });
+    })
+
+    test("POST 400: responds with error message when provided invalid username ", () => {
+      const newComment = {
+        username: "Peter",
+        body: "Test test test"
+      };
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(400)
+        .then(({ body }) => {
+          const { msg } = body;
+          expect(msg).toBe("Bad request");
+        });
+    });
+
+    test("POST 400: responds with error message when provided newComment without a body", () => {
+      const newComment = {
+        username: "rogersop"
+      };
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(400)
+        .then(({ body }) => {
+          const { msg } = body;
+          expect(msg).toBe("Comment missing username/body");
+        });
+    });
+  
+      test("POST 400: responds with error message when provided newComment with a article that does not exist ", () => {
+     const newComment = {
+       username: "rogersop",
+       body: "test body comment",
+     };    return request(app)
+          .post("/api/articles/999/comments")
+          .send(newComment)
+          .expect(400)
+          .then(({ body }) => {
+            const { msg } = body;
+            expect(msg).toBe("Bad request");
+          });
+      });
+  });
+  
+
```

Comment:
Excellent tests here Peter. A few more I think you should add here:

- a 201 that posts a comment when the body has extra properties. These extar properties should be ignored as long as username and body are present
- a 400 for when passed an invalid article id e.g. a string